src/stavehairpin.ts

```diff
@@ -165,23 +203,23 @@ export class StaveHairpin extends Element {
     ctx.closePath();
   }
 
-  draw() {
+  draw(): void {
     this.checkContext();
     this.setRendered();
 
     const firstNote = this.first_note;
     const lastNote = this.last_note;
+    if (!firstNote || !lastNote) throw new Vex.RERR('NoNote', 'Notes required to draw');
 
     const start = firstNote.getModifierStartXY(this.position, 0);
     const end = lastNote.getModifierStartXY(this.position, 0);
 
     this.renderHairpin({
       first_x: start.x,
       last_x: end.x,
-      first_y: firstNote.getStave().y + firstNote.getStave().height,
-      last_y: lastNote.getStave().y + lastNote.getStave().height,
-      staff_height: firstNote.getStave().height,
+      first_y: firstNote.checkStave().getY() + (firstNote.checkStave().getHeight() ?? 0),
```

Comment:
This `?? 0` here is a silent failure. It seems like `stave.height` should not be optional, all staves must have a height. Can we make it (and the result of `getHeight`) non-optional?

Reply:
yes it is always initialised in constructor, done